### PR TITLE
Indicate pending offers

### PR DESCRIPTION
**Pull request by michalsidzej**

### CODE REVIEW

File: packages/backend/src/core/AccountService.ts [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
     const id = await this.positionRepository.findIdByEthereumAddress(address)
-    const hasUpdates = false // TODO: find forced actions in progress
+
+    if (!id) {
+      return {
+        address,
+      }
+    }
+
+    const pendingCount = await this.offerRepository.countPendingByPositionIdA(
```
Comment (sz-piotr) [inline packages/backend/src/core/AccountService.ts:24]: In a separate PR we should also take transactions into account.

File: packages/backend/src/core/AccountService.ts [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+      id
+    )
     return {
       address,
       positionId: id,
-      hasUpdates,
+      hasUpdates: !!pendingCount,
```
Comment (sz-piotr) [inline packages/backend/src/core/AccountService.ts:30]: pendingCount > 0 would be clearer, but it's just a nitpick

File: packages/backend/src/peripherals/database/ForcedTradeOfferRepository.ts [hunk 3]
```diff
@@ -188,11 +188,22 @@ export class ForcedTradeOfferRepository extends BaseRepository {
     return rowIds.map(String).map(AssetId)
   }
 
-  async getPendingByPositionIdA(positionIdA: bigint) {
-    const rows = await this.knex('forced_trade_offers')
+  private getPendingByPositionIdAQuery(positionIdA: bigint) {
+    return this.knex('forced_trade_offers')
       .where({ position_id_a: positionIdA })
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/ForcedTradeOfferRepository.ts:193]: The positionId should be compared to both positionIdA and positionIdB

File: packages/backend/src/core/AccountService.ts (hunk 2 above)
Comment (sin6pi7) [inline packages/backend/src/core/AccountService.ts:30]:
```suggestion
      hasUpdates: pendingCount > 0,
```

That's more straightforward

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/frontend/src/pages/common/AccountDetails.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+import { stringAs, stringAsBigInt } from '@explorer/shared'
 import { EthereumAddress } from '@explorer/types'
 import { z } from 'zod'
 
 export type AccountDetails = z.infer<typeof AccountDetails>
 export const AccountDetails = z.object({
   address: stringAs(EthereumAddress),
-  positionId: z.bigint().optional(),
-  hasUpdates: z.boolean(),
+  positionId: stringAsBigInt().optional(),
```
Comment (sin6pi7) [inline packages/frontend/src/pages/common/AccountDetails.ts:8]: Why `stringAsBigInt`?

**Review by sin6pi7 — CHANGES_REQUESTED** [review]

File: packages/frontend/src/pages/common/AccountDetails.ts (hunk 4 above)
Comment (michalsidzej) [reply packages/frontend/src/pages/common/AccountDetails.ts:8]: We were getting a bug ```Uncaught ZodError: [
  {
    "code": "invalid_type",
    "expected": "bigint",
    "received": "string",
    "path": [
      "account",
      "positionId"
    ],
    "message": "Expected bigint, received string"
  }
]
    ```

File: packages/backend/src/peripherals/database/ForcedTradeOfferRepository.ts (hunk 3 above)
Comment (sin6pi7) [reply packages/backend/src/peripherals/database/ForcedTradeOfferRepository.ts:193]: @michalsidzej I assume this is going to be done in a separate PR?

Comment (michalsidzej) [reply packages/backend/src/peripherals/database/ForcedTradeOfferRepository.ts:193]: Yes, waiting for @sz-piotr design updates, there are few moving parts with changing this approach

**Review by sin6pi7 — APPROVED** [review]

**Pull request merged**